android/src/main/java/ie/gov/tracing/network/Fetcher.kt

```diff
@@ -270,35 +424,84 @@ class Fetcher {
                 Events.raiseEvent(Events.INFO, "triggerCallback - success")
                 SharedPrefs.setLong("notificationSent", System.currentTimeMillis(), context)
 
-            } catch(ex: Exception) {
+                saveMetric("CALLBACK_REQUEST", context)
+            } catch (ex: Exception) {
                 Events.raiseError("triggerCallback - error", ex)
             }
         }
 
+        @SuppressLint("CheckResult")
         @JvmStatic
         fun saveMetric(event: String, context: Context, payload: Map<String, Any>? = null) {
             try {
                 val analytics = SharedPrefs.getBoolean("analyticsOptin", context)
-                val version = Tracing.version().getString("display").toString()
+                val version = SharedPrefs.getString("version", context)
 
-                if(!analytics) {
+                if (!analytics) {
                     Events.raiseEvent(Events.INFO, "saveMetric - not saving, no opt in")
                     return
                 }
 
                 val metric = Metric("android", event, version, payload)
 
-                val success = post("/metrics", Gson().toJson(metric), context)
-
-                if (!success) {
-                    Events.raiseEvent(Events.ERROR, "saveMetric - failed: $event")
-                    return
+                Single.fromCallable {
```

Comment:
Do the metric post off the main thread, there seems to be a situation where android is killing it when it sends the DAILY_TRACE metric